store/copr/batch_coprocessor.go

```diff
@@ -523,24 +523,30 @@ func balanceBatchCopTask(ctx context.Context, kvStore *kvStore, originalTasks []
 	return ret
 }
 
-func buildBatchCopTasks(bo *backoff.Backoffer, store *kvStore, ranges *KeyRanges, storeType kv.StoreType, mppStoreLastFailTime map[string]time.Time, ttl time.Duration, balanceWithContinuity bool, balanceContinuousRegionCount int64) ([]*batchCopTask, error) {
+func buildBatchCopTasks(bo *backoff.Backoffer, store *kvStore, rangess []*KeyRanges, storeType kv.StoreType, mppStoreLastFailTime map[string]time.Time, ttl time.Duration, balanceWithContinuity bool, balanceContinuousRegionCount int64, partitionIDs []int64) ([]*batchCopTask, error) {
 	cache := store.GetRegionCache()
 	start := time.Now()
 	const cmdType = tikvrpc.CmdBatchCop
-	rangesLen := ranges.Len()
+	rangesLen := 0
+
 	for {
-		locations, err := cache.SplitKeyRangesByLocations(bo, ranges)
-		if err != nil {
-			return nil, errors.Trace(err)
-		}
 		var tasks []*copTask
-		for _, lo := range locations {
-			tasks = append(tasks, &copTask{
-				region:    lo.Location.Region,
-				ranges:    lo.Ranges,
-				cmdType:   cmdType,
-				storeType: storeType,
-			})
+		rangesLen = 0
+		for i, ranges := range rangess {
+			rangesLen += ranges.Len()
+			locations, err := cache.SplitKeyRangesByLocations(bo, ranges)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			for _, lo := range locations {
+				tasks = append(tasks, &copTask{
+					region:      lo.Location.Region,
+					ranges:      lo.Ranges,
+					cmdType:     cmdType,
+					storeType:   storeType,
+					partitionID: int64(i),
```

Comment:
This indicates partition id is always `0 - len(rangess)`?

Reply:
Here is only the index of `partitionId` representing the partition data stored in this region. Probably better to rename to partitionIndex.
The reason for this is that it is more convenient to build tableRegions later....